app.js

```diff
@@ -1,180 +1,188 @@
+
+  if (branch) { args.push("-b", branch); }
+
+  if (privateKey && !(username && password)) { // if provided ssh Key and not username and password
+    const [newGitKey, sshCmd] = await getSSHCommand(privateKey, saveCreds);
+    gitKey = newGitKey;
+    args.push(`--config core.sshCommand="${sshCmd}"`);
+  }
+
+  if (extraArgs) { args.push(...parsers.array(extraArgs)); }
+  args.push(path);
+
+  // clone using key file
+  let didTurnAgentUp = false;
+  try {
+    if (gitKey) { didTurnAgentUp = await turnSshAgentUp(gitKey); }
+    // run clone
+    const cloneResult = await execGitCommand(args);
+    return cloneResult;
+  } finally {
+    if (didTurnAgentUp) { await killSshAgent(); }
+    if (privateKey && gitKey) {
+      await gitKey.dispose();
+    }
+  }
+}
+
+async function clonePublic(action) {
+  await verifyGitVersion();
+  const path = parsers.path(action.params.path);
+  const repo = parsers.string(action.params.repo);
+  const branch = parsers.string(action.params.branch);
+  const { overwrite, extraArgs } = action.params;
+
+  if (!repo.startsWith("https://")) {
+    throw new Error("Please use HTTPS format URL for public repositories. Anonymous SSH is not supported in method \"Clone public repository\".");
+  }
+
+  if (overwrite) { await tryDelete(path); }
+
+  const args = ["clone", repo];
+
+  if (branch) { args.push("-b", branch); }
+
+  if (extraArgs) { args.push(...parsers.array(extraArgs)); }
+  args.push(path);
+
+  const cloneResult = await execGitCommand(args);
+  return cloneResult;
+}
+
+async function pull(action) {
+  const path = parsers.path(action.params.path);
+  const { force, commitMerge, extraArgs } = action.params;
+  if (!path) { throw new Error("Repository Path is a requried parameter."); }
+  const didTurnAgentUp = isWin ? false : await turnSshAgentUp(await GitKey.fromRepoFolder(path));
+  const args = ["pull"];
+  if (force) { args.push("-f"); }
+  args.push(`--${commitMerge ? "" : "no-"}commit`);
+  args.push(...parsers.array(extraArgs));
+  try {
+    const result = await execGitCommand(args, path);
+    return result;
+  } finally {
+    if (didTurnAgentUp) { await killSshAgent(); }
+  }
+}
+
+async function pushTag(action, settings) {
+  const path = parsers.path(action.params.path);
+  const tagName = parsers.string(action.params.tagName);
+  const message = parsers.string(action.params.message);
+  const push = !action.params.noPush;
+  const username = parsers.string(action.params.username || settings.username);
+  const email = parsers.string(action.params.email || settings.email);
+
+  let didTurnAgentUp = false;
+
+  if (!path || !tagName) { throw new Error("Both Repository Path and Tag Name are required parameters."); }
+
+  await setUsernameAndEmail(username, email, path);
+
+  if (push && !isWin) {
+    didTurnAgentUp = await turnSshAgentUp(await GitKey.fromRepoFolder(path));
+  }
+
+  const tagArgs = ["tag"];
+  if (message) {
+    tagArgs.push("-a", tagName, `-m "${message}"`);
```

Comment:
We can separate the last two arguments here, just like above: ```tagArgs.push("-a", tagName, "-m", message)```. It will also look cleaner.